supplier_user_stock_data_purchase/models/supplier_stock.py

```diff
@@ -1,11 +1,86 @@
 # Copyright 2020 Quartile Limited
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
 
-from odoo import fields, models
+import odoo.addons.decimal_precision as dp
+from odoo import api, fields, models
 
 
 class SupplierStock(models.Model):
     _inherit = "supplier.stock"
 
     website_published = fields.Boolean("Visible in Portal / Website", copy=False,)
     readonly_record = fields.Boolean("Readonly Record", copy=False, default=False,)
+    website_quantity = fields.Selection(
+        [("1", "1"), ("2", "2"), ("3", ">=3")],
+        string="Webiste Quantity",
+        readonly=True,
+    )
+    retail_chf = fields.Float(
+        related="product_id.product_tmpl_id.retail_chf",
+        string="Retail CHF",
+        digits=dp.get_precision("Product Price"),
+    )
+    retail_eur = fields.Float(
+        related="product_id.product_tmpl_id.retail_eur",
+        string="Retail EUR",
+        digits=dp.get_precision("Product Price"),
+    )
+    retail_usd = fields.Float(
+        related="product_id.product_tmpl_id.retail_usd",
+        string="Retail USD",
+        digits=dp.get_precision("Product Price"),
+    )
+    retail_rmb = fields.Float(
+        related="product_id.product_tmpl_id.retail_rmb",
+        string="Retail RMB",
+        digits=dp.get_precision("Product Price"),
+    )
+
+    cost_hkd = fields.Float(
+        related="product_id.product_tmpl_id.net_price",
+        string="My Cost in HKD",
+        digits=dp.get_precision("Product Price"),
+        store=True,
+    )
+    cost_chf = fields.Float(
+        string="My Cost in CHF",
+        digits=dp.get_precision("Product Price"),
+        compute="_compute_cost_price",
+        store=True,
+    )
+    cost_eur = fields.Float(
+        string="My Cost in EUR",
+        digits=dp.get_precision("Product Price"),
+        compute="_compute_cost_price",
+        store=True,
+    )
+    cost_usd = fields.Float(
+        string="My Cost in USD",
+        digits=dp.get_precision("Product Price"),
+        compute="_compute_cost_price",
+        store=True,
+    )
+    special_offer_currency_id = fields.Many2one("res.currency")
+
+    @api.multi
+    @api.depends("cost_hkd")
+    def _compute_cost_price(self):
```

Comment:
Just a style preference. I'd insert `currency = self.env.user.company_id.currency_id` before the loop to make the following part a bit more concise.